### PR TITLE
test(validation): add GovernedByOptimismTest

### PR DESCRIPTION
**Pull request by edobry**
### Description

This PR adds the `GovernedByOptimismTest` test to the `validation` package, which ensures that any chain which has the `governed_by_optimism` field set to `true` in its config also has its `ProxyAdmin` address set to the Optimism multisig.

This change was broken out into a separate PR from #718 to allow for the `validation/go.mod` file to reference the new `superchain` module version.

### Metadata

- **In service of**: #676 
- **Depends on:**
	- #717 
	- #718

### CODE REVIEW

**Comment by geoknee** [comment]
> This change was broken out into a separate PR from #718 to allow for the `validation/go.mod` file to reference the new `superchain` module version.

I'm not sure I follow you. IIRC because this repo uses a `go.work` file, all the modules should reference the local files anyway.

**Comment by geoknee** [comment]
See my comment here https://github.com/ethereum-optimism/superchain-registry/pull/718#issuecomment-2505848103 we already have a "key handover test"  and I would like to understand how this new test relates to that one.

File: validation/governed-by-optimism_test.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func testGovernedByOptimism(t *testing.T, chain *ChainConfig) {
+	chainID := chain.ChainID
+	superchain := OPChains[chainID].Superchain
+
+	optimismMultisig := standard.Config.MultisigRoles[superchain].KeyHandover.L1.Universal["ProxyAdmin"]["owner()"]
+
+	if chain.GovernedByOptimism == true {
+		require.Equal(t, chain.Addresses.ProxyAdmin.String(), optimismMultisig, "Chains using Optimism governance must have their ProxyAdmin set to the Optimism multisig")
```
Comment (geoknee) [inline validation/governed-by-optimism_test.go:21]: Possibly related to my previous comment, you currently have a test failure here https://app.circleci.com/pipelines/github/ethereum-optimism/superchain-registry/10103/workflows/6ad5f5b4-8e89-4d1e-868d-14043b8fbc96/jobs/45810 suggesting that OPM is not governed by optimism :-) 

The existing test defined here https://github.com/ethereum-optimism/superchain-registry/blob/c56b99231b9033d4db7d64f4fe4eeca72d733a8a/validation/standard/standard-config-roles-mainnet.toml#L9-L11 is passing when I run `just validate 10`.

Comment (geoknee) [inline validation/governed-by-optimism_test.go:21]: 1. You should permute the expected and actual slots for `require.Equal`
2. You need to use `ProxyAdminOwner` not `ProxyAdmin`		

```suggestion
		require.Equal(t, optimismMultisig, chain.Addresses.ProxyAdminOwner.String(), "Chains using Optimism governance must have their ProxyAdminOwner set to the Optimism multisig")
```

**Comment by edobry** [comment]
> > This change was broken out into a separate PR from #718 to allow for the `validation/go.mod` file to reference the new `superchain` module version.
> 
> I'm not sure I follow you. IIRC because this repo uses a `go.work` file, all the modules should reference the local files anyway.

good point, I wasn't familiar with `go.work`, I'll merge the PRs to reflect this.

**Pull request closed without merging**